tests/integ/conftest.py

```diff
@@ -63,26 +79,71 @@ def nonce(request):
     return request.config.getoption("--nonce")
 
 
-@pytest.yield_fixture(autouse=True)
-def cleanup_objects(engine):
-    yield
-
-    # TODO track bound models w/model_bound signal (TODO), then use boto3 to scan/delete by Meta.table_name
-    # Running tests individually may break if the User table isn't bound as part of that test
-    users = list(engine.scan(User))
-    engine.delete(*users)
-
-
 @pytest.fixture(scope="session")
-def dynamodb():
-    return fixed_ddb_client
+def dynamodb(do_provide_aws):
+    return fixed_session.client("dynamodb", endpoint_url=resource_options['endpoint_url'])
 
 
 @pytest.fixture(scope="session")
-def dynamodbstreams():
-    return fixed_streams_client
+def dynamodbstreams(do_provide_aws):
+    return fixed_session.client("dynamodbstreams", endpoint_url=resource_options['endpoint_url'])
 
 
 @pytest.fixture
 def engine(dynamodb, dynamodbstreams):
-    return Engine(dynamodb=dynamodb, dynamodbstreams=dynamodbstreams)
+    yield Engine(dynamodb=dynamodb, dynamodbstreams=dynamodbstreams)
+
+
+def get_open_port():
+    with contextlib.closing(
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(("", 0))
+        s.listen(1)
+        port = s.getsockname()[1]
+        return port
+
+
+def get_ddb_local():
+    localdir = pytest.config.rootdir.join('.dynamodb-local').strpath
+    if not os.path.exists(localdir):
+        tempdir = localdir + '.tmp'
+        if os.path.exists(tempdir):
+            shutil.rmtree(tempdir)
+        os.mkdir(tempdir)
+        r = requests.get(
+            'https://s3-us-west-2.amazonaws.com/' +
+            'dynamodb-local/dynamodb_local_latest.zip',
+            stream=True)
+        dist = os.path.join(tempdir, 'dist.zip')
+        with open(dist, 'wb') as f:
+            for chunk in r.iter_content(chunk_size=1024):
+                if chunk:
+                    f.write(chunk)
+        zip_ref = zipfile.ZipFile(dist, 'r')
+        zip_ref.extractall(tempdir)
+        zip_ref.close()
+        os.rename(tempdir, localdir)
+    return localdir
+
+
+def start_dynamodb_local():
+    cwd = get_ddb_local()
+    port = get_open_port()
```

Comment:
Why does `resource_options` have a port if we use the value immediately to construct the endpoint_url?  Should the dict be updated with the value we get back from `get_open_point()`?

Reply:
Good point